File: backend-nestjs/patient-service/src/knowledge/entities/document.entity.ts
import { Entity, Column, PrimaryGeneratedColumn } from 'typeorm';

@Entity('documents')
export class DocumentEntity {
  @PrimaryGeneratedColumn({ type: 'bigint' })
  id: number;

  @Column({ name: 'tenant_id', type: 'bigint' })
  tenantId: number;

  @Column({ name: 'kb_id', type: 'bigint' })
  kbId: number;

  @Column({ name: 'org_id', type: 'bigint' })
  orgId: number;

  @Column({ name: 'uploader_id', type: 'bigint' })
  uploaderId: number;

  @Column({ name: 'file_name' })
  fileName: string;

  @Column({ name: 'file_type', nullable: true })
  fileType: string;

  @Column({ name: 'file_size', nullable: true })
  fileSize: number;

  @Column({ name: 'minio_url' })
  minioUrl: string;
}

File: backend-java/cdm-ai/src/main/java/com/cdm/ai/entity/DocumentEntity.java
package com.cdm.ai.entity;

import com.baomidou.mybatisplus.annotation.TableName;
import com.baomidou.mybatisplus.annotation.TableField;
import com.cdm.common.domain.BaseEntity;
import com.cdm.ai.vo.DocumentVo;
import lombok.Getter;
import lombok.Setter;
import lombok.NoArgsConstructor;
import java.time.LocalDateTime;

@TableName("documents")
@Getter
@Setter
@NoArgsConstructor
public class DocumentEntity extends BaseEntity {

    private Long tenantId;
    private Long kbId;
    private Long orgId;
    private Long uploaderId;
    private String fileName;
    private String fileType;
    private Integer fileSize;
    private String minioUrl;
    private LocalDateTime updatedAt;

    @TableField(exist = false)
    private String status = "completed"; 
    
    @TableField(exist = false)
    private Integer chunkCount = 0;

    public static DocumentVo toVo(DocumentEntity entity) {
        if (entity == null) return null;
        return DocumentVo.builder()
                .id(entity.getId())
                .tenantId(entity.getTenantId())
                .kbId(entity.getKbId())
                .orgId(entity.getOrgId())
                .uploaderId(entity.getUploaderId())
                .fileName(entity.getFileName())
                .fileType(entity.getFileType())
                .fileSize(entity.getFileSize())
                .minioUrl(entity.getMinioUrl())
                .status(entity.getStatus())
                .chunkCount(entity.getChunkCount())
                .updatedAt(entity.getUpdatedAt())
                .createdAt(entity.getCreatedAt())
                .build();
    }
}
